### ISSUE TITLE
Feature request: CLI flag to fail (exit non-zero) if parsing errors occur

### ISSUE THREAD

**Issue by bryanburke** [bridgecrewio/checkov#5248]
**Describe the issue**

Currently, Checkov exits zero (indicating success) if any file parsing errors occur. This behavior makes integrations with CI pipelines more difficult and can introduce false positives. For example, an IaC template containing actual security flaws may slip past the CI process unnoticed if Checkov encounters an error parsing the file for any reason.

**Feature request**

I would therefore like to request a new CLI flag such as `--fail-on-parsing-error` or similar (with a corresponding `fail-on-parsing-error: true` option in the config file) to instruct Checkov to exit non-zero if any file parsing errors occur. Outputting additional information about the parsing failure (such as which framework Checkov tried to use to parse the file) would also be extremely helpful.

**Current behavior**

The current behavior acts as in this example:

```sh
$ checkov -d .
[...]
Passed checks: 0, Failed checks: 0, Skipped checks: 0, Parsing errors: 1

Error parsing file foobar.json
$ echo $?
0
```

**Desired behavior**

The desired behavior might act as in this example:

```sh
$ checkov -d .
[...]
Passed checks: 0, Failed checks: 0, Skipped checks: 0, Parsing errors: 1

Error parsing file foobar.json which Checkov identified as appropriate for the "arm" framework
$ echo $?
1
```

Thank you for your time!
